### ISSUE TITLE
brscan5's udev rule contains deprecated key "SYSFS"

### ISSUE THREAD

**Issue by leana8959** [NixOS/nixpkgs#332307]
### Describe the bug
When using the `hardware.sane.brscan5` option, the udev rule supplied by the upstream contains the deprecated key "SYSFS".

This cause `systemd.udevd.service` to fail.

```log
Aug 04 19:57:29 carbon systemd-udevd[2092554]: /nix/store/dknjp9p11iwwjxqywhs8cpy9ss2gj0qs-udev-rules/49-brother-mfp-brscan5-1.0.2-2.rules:17 Invalid key 'SYSFS'.
``` 

### Steps To Reproduce
First set
```nix
hardware.sane = {
  enable = true;
  brscan5.enable = true;
};
```
Then run 
```bash
sudo systemd status systemd-udevd.service
```

### Expected behavior
`brscan5`'s udev rule shouldn't fail `systemd-udevd.service`.

### Additional context
I did a quick grep on the latest (brscan5-1.3.5-0.amd64.deb) upstream package, the offending "SYSFS" key is still present. Maybe we need to patch this ourselves for now ?

### Notify maintainers
@mattchrist

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.43, NixOS, 24.05 (Uakari), 24.05.20240730.cf05eea`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.21.4`
 - nixpkgs: `/nix/store/z3vrsz16ypv60m2pp0gi7x7gpx40gfh1-source`
```
---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by mattchrist** [comment]
I don't believe this is causing systemd-udev to fail, but does cause an error message in the logs.

On my system, `systemctl status systemd-udevd` shows the service is `active (running)`. I do see those errors in the journal as well.

**Comment by leana8959** [comment]
Thank you for your quick reply !

In fact, I noticed that when I run `scanimage -L`, my scanner isn't detected. However, running the same command with sudo works. I suspect that there might be something wrong with the udev file, I filed the bug report when I saw that it's not recognizing the SYSFS key.

Is it expected for `scanimage -L` to only work when run as root or is it somehow related to this issue ?

Thanks :)

**Comment by eclairevoyant** [comment]
You probably need the `scanner` and possibly `lp` group as stated in the desc for `hardware.sane.enable`

**Comment by zoenglinghou** [comment]
In the latest `nixos-unstable`, the `udev` rules builder verifies the rules w/ `udevadm`, and it causes the build to fail
```
       > Checking that all programs called by absolute paths in udev rules exist... OK
       > Verifying udev rules using udevadm verify...
       > /nix/store/r04cgapxy0xhwjfbvmw7rxzvc5rm1yzn-udev-rules/49-brother-mfp-brscan5-1.0.2-2.rules:17 Invalid key 'SYSFS'.
       > /nix/store/r04cgapxy0xhwjfbvmw7rxzvc5rm1yzn-udev-rules/49-brother-mfp-brscan5-1.0.2-2.rules: udev rules check failed.
```

**Comment by Chaostheorie** [comment]
For anyone else who is blocked by this when testing 25.05, the below overlay will work around this issue by replacing all `SYSFS` keys with `ATTR`:

```nix
  # override brscan 5 to work around nixpkgs#332307
  nixpkgs.overlays = [
    (_: super: {
      brscan5 = super.brscan5.overrideAttrs (
        _: previousAttrs: {
          postPatch =
            (lib.optionalString (previousAttrs ? "postPatch") previousAttrs.postPatch)
            + ''
              sed -i 's/SYSFS/ATTR/g' opt/brother/scanner/brscan5/udev-rules/*.rules
            '';
        }
      );
    })
  ];
```

**Comment by mattchrist** [comment]
Here's the open MR that fixes this. https://github.com/NixOS/nixpkgs/pull/332311

It's reviewed already. I think it only needs someone to merge it.
